cli/ioctl/cmd/action/action.go

```diff
@@ -40,9 +51,52 @@ var ActionCmd = &cobra.Command{
 func init() {
 	ActionCmd.AddCommand(actionHashCmd)
 	ActionCmd.AddCommand(actionTransferCmd)
+	ActionCmd.AddCommand(actionDeployCmd)
+	ActionCmd.AddCommand(actionInvokeCmd)
+	setActionFlags(actionTransferCmd, actionDeployCmd, actionInvokeCmd)
+}
+
+func setActionFlags(cmds ...*cobra.Command) {
+	for _, cmd := range cmds {
+		cmd.Flags().Uint64VarP(&gasLimit, "gas-limit", "l", 0, "set gas limit")
+		cmd.Flags().Int64VarP(&gasPrice, "gas-price", "p", 0, "set gas prize")
+		cmd.Flags().StringVarP(&alias, "alias", "a", "", "choose signing key")
+		cmd.MarkFlagRequired("gas-limit")
+		cmd.MarkFlagRequired("gas-price")
+		cmd.MarkFlagRequired("alias")
+		if cmd == actionDeployCmd || cmd == actionInvokeCmd {
+			cmd.Flags().BytesHexVarP(&bytecode, "bytecode", "b", nil, "set the byte code")
+			actionInvokeCmd.MarkFlagRequired("bytecode")
+		}
+	}
 }
 
-func sendAction(request *iotexapi.SendActionRequest) string {
+func sendAction(elp action.Envelope) string {
+	fmt.Printf("Enter password #%s:\n", alias)
+	bytePassword, err := terminal.ReadPassword(syscall.Stdin)
+	if err != nil {
+		log.L().Error("fail to get password", zap.Error(err))
```

Comment:
Will you duplicate the message because of this log and then returning error message to the caller.

Reply:
They are different errors. The err from ReadPassword may be hard to read.